jaxopt/_src/osqp.py

```diff
@@ -599,3 +597,187 @@ def __post_init__(self):
       self.check_primal_dual_infeasability = not jit
 
     self.optimality_fun = _make_osqp_optimality_fun(self.matvec_Q, self.matvec_A)
+
+
+class OSQP_to_BoxOSQP:
+  """Convert instance of ``OSQP`` (which is ``CVXPY_QP`` compatible) to BoxOSQP."""
+
+  @staticmethod
+  def transform_matvec(matvec_Q, matvec_A, matvec_G):
+    """Return matvec_Q, matvec_A of BoxOSQP from the matvec_Q, matvec_A, matvec_G of OSQP."""
+    if matvec_A is None and matvec_G is None:
+      matvec_A_box = None
+    elif matvec_A is None:
+      def matvec_A_box(params, x):
+        A, params_G = params
+        return DenseLinearOperator(A)(x), matvec_G(params_G, x)
+    elif matvec_H is None:
+      def matvec_A_box(params, x):
+        params_A, G = params
+        return matvec_A(params_A, x), DenseLinearOperator(G)(x)
+    else:
+      def matvec_A_box(params, x):
+        params_A, params_H = params
+        return matvec_A(params_A, x), matvec_G(params_G, x)
+    return matvec_Q, matvec_A_box
+  
+  @staticmethod
+  def fit_transform(params: Optional[jnp.ndarray],
```

Comment:
Can't we just call it `transform`? `fit_transform` would mean that we store some stuff in the object.